ferry/config.py

```diff
@@ -23,9 +26,15 @@
 # The latter three settings can also be set to a function. The function should return the
 # specified setting when called.
 CAS_USE_COOKIE = True
-CAS_CREDENTIAL_NETID = functools.lru_cache(lambda: input("Yale NetId: "))
-CAS_CREDENTIAL_PASSWORD = functools.lru_cache(getpass.getpass)
-CAS_COOKIE_CASTGC = functools.lru_cache(lambda: input("CASTGC Cookie: "))
+CAS_CREDENTIAL_NETID: "_lru_cache_wrapper[str]" = functools.lru_cache(
```

Comment:
This annotation seems kinda ugly - can we not just make it a `Callable[[], str]`?

Reply:
This type produces an error when it's overridden by config_private - for instance, with the cookie we have
```
ferry/config.py:43: error: Incompatible import of "CAS_COOKIE_CASTGC" 
(imported name has type "str", local name has type "Callable[[], str]")
```

Perhaps `Union[Callable[[], str], str]`?

Reply:
Would rather that we change config_private - it's a bit more ugly on the config side but will make usage cleaner
```python
# config_private.py
CAS_COOKIE_CASTGC = lambda: "some secret value"
```